components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,199 @@
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager compManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(compManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.patientsResource = (DefaultPatientsResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.queries = this.mocker.getInstance(QueryManager.class);
+        this.uri = new URI("http://uri");
+
+        doReturn(this.uri).when(this.uriInfo).getBaseUri();
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+        ReflectionUtils.setFieldValue(this.patientsResource, "uriInfo", this.uriInfo);
+
+        doReturn("00000001").when(this.patient).getId();
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
+    }
+
+    @Test
+    public void addPatientUserDoesNotHaveAccess() throws XWikiRestException {
+        WebApplicationException exception = new WebApplicationException();
+        doReturn(false).when(this.access).hasAccess(Right.EDIT, null, mock(EntityReference.class));
+        try {
+            Response response = this.patientsResource.addPatient("");
+        }
+        catch (WebApplicationException ex){
+            exception = ex;
+        }
+        Assert.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), exception.getResponse().getStatus());
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", "");
+    }
+
+    @Test
+    public void addNullPatient() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        Response response = this.patientsResource.addPatient(null);
+        Assert.assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+        verify(this.logger).error(eq("Could not process remote matching request: {}"), anyString(), anyObject());
+    }
+
+    @Test
+    public void addPatientAsJSON() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        JSONObject json = new JSONObject();
+        Response response = this.patientsResource.addPatient(json.toString());
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", json.toString());
+    }
```

Comment:
The only thing you are currently verifying in this test is a logger debug statement that will execute every time whenever the method is called. There is a lot of functionality that you are missing here. 
You need to mock the behaviour of `this.repository.createNewPatient()` and verify that that call occurred. As you have it written currently that method call will return null, and the following statement `patient.updateFromJSON(jsonInput);` will generate a `NullPointerException`. Once that change is made you should  perform some checks on the return value of the `addPatient` method in this test